query.go

```diff
@@ -0,0 +1,104 @@
+package loki
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+type QueryType int
+
+const (
+	InstantQuery QueryType = iota
+	RangeQuery
+	LabelsQuery
+	LabelValuesQuery
+)
+
+func (t QueryType) Endpoint() string {
+	switch t {
+	case InstantQuery:
+		return "/loki/api/v1/query"
+	case RangeQuery:
+		return "/loki/api/v1/query_range"
+	case LabelsQuery:
+		return "/loki/api/v1/labels"
+	case LabelValuesQuery:
+		return "/loki/api/v1/label/%s/values"
+	default:
+		return ""
+	}
+}
+
+type Direction int
+
+const (
+	Undefined Direction = iota
+	Backward
+	Forward
+)
+
+func (d Direction) String() string {
+	switch d {
+	case Forward:
+		return "forward"
+	case Backward:
+		return "backward"
+	default:
+		return ""
+	}
+}
+
+// Query contains all necessary fields to execute instant and range queries and print the results.
+type Query struct {
+	Type        QueryType
+	QueryString string
+	Start       time.Time
+	End         time.Time
+	Limit       int
+	Step        int
+	Interval    int
+	Direction   Direction
+	PathParams  []interface{}
+}
+
+func (q *Query) Endpoint() string {
+	return fmt.Sprintf(q.Type.Endpoint(), q.PathParams...)
+}
+
+func (q *Query) Values() url.Values {
+	v := url.Values{}
+	if q.QueryString != "" {
+		v.Set("query", q.QueryString)
+	}
+	if q.Start.Unix() > 0 {
```

Comment:
if it's and instant query it should not send start and end, but instead `time`. see https://grafana.com/docs/loki/latest/api/#get-lokiapiv1query